views/default/input/autocomplete.php

```diff
@@ -35,17 +35,15 @@
 	$params['match_owner'] = $vars['match_owner'];
 	unset($vars['match_owner']);
 }
-$ac_url_params = http_build_query($params);
+$vars['type'] = 'text';
+$vars['data-source'] = elgg_get_site_url() . 'livesearch?' . http_build_query($params);
 
-elgg_load_js('elgg.autocomplete');
-elgg_load_js('jquery.ui.autocomplete.html');
+echo elgg_format_element('input', $vars);
 
+// inline script in case loaded via ajax
 ?>
-
 <script>
-require(['elgg'], function (elgg) {
-	elgg.provide('elgg.autocomplete');
-	elgg.autocomplete.url = "<?php echo elgg_get_site_url() . 'livesearch?' . $ac_url_params; ?>";
+require(['elgg/autocomplete'], function (autoc) {
+	autoc.init();
```

Comment:
I have stopped using `elgg_require_js()`, as I operate under an assumption
that any view can potentially be loaded via ajax. We could do an audit of
all require calls to see how likely the views are to be loaded via ajax.
On Nov 16, 2015 3:11 PM, "Steve Clay" <notifications@github.com> wrote:

> In views/default/input/autocomplete.php
> <https://github.com/Elgg/Elgg/pull/9125#discussion_r44926617>:
>
> >  <script>
> > -require(['elgg'], function (elgg) {
> > -	elgg.provide('elgg.autocomplete');
> > -	elgg.autocomplete.url = "<?php echo elgg_get_site_url() . 'livesearch?' . $ac_url_params; ?>";
> > +require(['elgg/autocomplete'], function (autoc) {
> > +	autoc.init();
>
> Are there other initialization bugs we could fix using this pattern? I
> think it'd be worth delaying 2.0 a bit if we could knock them all out.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/9125/files#r44926617>.
>